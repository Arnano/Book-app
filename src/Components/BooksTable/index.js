import React from 'react';
import PropTypes from 'prop-types';
import StyledBooksTable from './style';

const BooksTable = ({ classes, children }) =>
  <StyledBooksTable className={classes}>
    {children}
  </StyledBooksTable>

BooksTable.propTypes = {
  classes: PropTypes.string,
  children: PropTypes.node
}

export default BooksTable;
