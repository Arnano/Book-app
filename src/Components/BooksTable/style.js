import Styled from 'styled-components';
import { grey } from '../../Style/colors';

const dropShadow = '0 2px 5px 0 rgba(0, 0, 0, 0.05)';

const StyledBooksTable = Styled.ul `
  max-width: 960px;
  margin: 10px auto;
  position: relative;
  border-radius: 8px;
  background-clip: padding-box;
  border: 1px solid ${grey};
  box-shadow: ${dropShadow};
  transition: box-shadow 0.3s ease;
  padding: 0;

  a > &:hover {
    cursor: pointer;
    box-shadow: 0 2px 10px 0 rgba(0, 0, 0, 0.15);
  }
`;

export default StyledBooksTable;