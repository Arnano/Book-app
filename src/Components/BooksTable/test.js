import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BooksTable from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Books table', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BooksTable>Table</BooksTable>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<BooksTable>Table</BooksTable>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
