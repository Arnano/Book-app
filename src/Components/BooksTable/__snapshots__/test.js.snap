// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books table has a valid snapshot 1`] = `
<ul
  className="sc-bdVaJa bAWDyw"
>
  Table
</ul>
`;
